Add simple debug logging to PythonUDF

Yicong-Huang opened the pull request:
This PR adds debug logging to PythonUDF. It is currently a simple implementation.

- The root logger is configured to have a StreamHandler and a FileHandler. The StreamHandler outputs to stderr and inherited by Java process (worker). The FileHander outputs to a log file.
- Python log configurations can be configured in `conf/web.conf`. By default, 
    - fileHandler log to files in `/tmp/` directory, and the log level is set to be `INFO`.
    - streamHandler log to stdout/stderr, and the log level is set to be `ERROR`.
- A log file will be created for each Python process. When python is running in parallel mode, there will be multiple log files spawned. The file name is `texera-python_udf-{time}-{pid}.log` format.
- Add `log_exception` decorator to log out all exceptions during defined PythonUDF operator APIs, such as `open`, `accept`, `close`, etc.
- Logger will now output class name, method name, and line number.
- Extracted operator base classes to make library dependency isolated.

<sup> Created from JetBrains using [CodeStream](https://codestream.com/?utm_source=cs&utm_medium=pr&utm_campaign=github*com)</sup>

avinash0161 commented on core/amber/src/main/resources/python_udf/demo_non_map_udf.py:
```diff
@@ -1,21 +1,29 @@
 class DemoOperator(texera_udf_operator_base.TexeraUDFOperator):
+    logger = logging.getLogger("PythonUDF.DemoOperator")
+
     def __init__(self):
```
why arent we putting the decorator on init?

Yicong-Huang replied:
added.

avinash0161 commented on core/amber/src/main/resources/python_udf/svm_classifier.py:
```diff
@@ -7,13 +8,15 @@
 class SVMClassifier(texera_udf_operator_base.TexeraMapOperator):
+    logger = logging.getLogger("PythonUDF.SVMClassifier")
 
     def __init__(self):
```
here too.

Yicong-Huang replied:
added.

avinash0161 commented on core/amber/src/main/resources/python_udf/texera_udf_server_main.py:
```diff
@@ -159,19 +172,67 @@ def _output_data(self):
+        # create a file handler
+
+        log_dir = args[0]
+        file_name = f"{datetime.utcnow().isoformat()}-{os.getpid()}"
```
please change

Yicong-Huang replied:
changed to `texera-python_udf-{time}-{pid}.log`

avinash0161 commented on core/conf/web.conf:
```diff
@@ -8,8 +8,15 @@
+	    	# python3 log diretory
+	        "dir": "/tmp/",
+	        # python3 log level
+	        "level": "ERROR"
```
lets change this to INFO or whatever is the least. by default lets log everything

Yicong-Huang replied:
changed.

avinash0161 reviewed (approved):
left some comments

Yicong-Huang commented:
@avinash0161 I added some more changes:
1. logger will now output class name, method name, and line number.
2. extracted operator base classes to make library dependency isolated.
Please give another round of review.

Also it is possible and doable to redirect python's print to log files: simply overrites `sys.stdout` to `logger.info`. However I think this is pretty hacky way to do so. PyFlink uses recommended users to [use log.info instead of print](https://ci.apache.org/projects/flink/flink-docs-release-1.12/dev/python/debugging.html#logging-infos), so I suggest we keep it this way.
